servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/LoadBalancerObserverFactory.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+/**
+ * Factory of {@link LoadBalancerObserver} instances.
+ */
+public interface LoadBalancerObserverFactory {
```

Comment:
Its worth debating on whether we want to create a real type here or just use `Fuction<String, LoadBalancerObserver>`. I like the interface because it gives a place to document what the parameter will be, but we could arguably do that in the `loadBalancerObserver(Fuction<String, LoadBalancerObserver>)` method. Opinions welcome.

Reply:
Either way works for me, in this case it's likely gonna be a lambda anyways.
Consider adding `@FunctionalInterface` annotation